.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,115 @@
+name: Build and Test
+on: push
+      
+jobs:  
+  setup_build_ext:
+    name: Run setup build_ext
+    runs-on: ubuntu-latest
+    container: benjaminrueth/precice
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+      - name: Install & upgrade pip3
+        run: |
+          apt-get install -y python3-pip
+          pip3 install --upgrade --user pip
+      - name: Install dependencies
+        run: |
+          pip3 install --user toml
+          python3 -c 'import toml; c = toml.load("pyproject.toml"); print("\n".join(c["build-system"]["requires"]))' | pip3 install -r /dev/stdin
+      - name: Run setup.py build_ext
+        run:  python3 setup.py build_ext
+
+  setup_install:
+    name: Run setup install
+    runs-on: ubuntu-latest
+    container: benjaminrueth/precice
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+      - name: Install & upgrade pip3
+        run: |
+          apt-get install -y python3-pip
+          pip3 install --upgrade --user pip
+      - name: Install dependencies
+        run: |
+          pip3 install --user toml
+          python3 -c 'import toml; c = toml.load("pyproject.toml"); print("\n".join(c["build-system"]["requires"]))' | pip3 install -r /dev/stdin
+      - name: Run setup install
+        run:  python3 setup.py install --user
+      - name: Test install
+        run:  python3 -c "import precice"
+        
+  setup_install_single_version_externally_managed:
+    name: Run setup install --single-version-externally-managed (for spack)
+    runs-on: ubuntu-latest
+    container: benjaminrueth/precice
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+      - name: Install & upgrade pip3
+        run: |
+          apt-get install -y python3-pip
+          pip3 install --upgrade --user pip
+      - name: Install dependencies
+        run: |
+          pip3 install --user toml
+          python3 -c 'import toml; c = toml.load("pyproject.toml"); print("\n".join(c["build-system"]["requires"]))' | pip3 install -r /dev/stdin
+      - name: Run setup install --single-version-externally-managed
+        run:  python3 setup.py install --single-version-externally-managed --root=/
+        
+  setup_test:
+    name: Run setup test
+    runs-on: ubuntu-latest
+    container: precice/ci-ubuntu-1804
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+      - name: Install & upgrade pip3
+        run: |
+          apt-get update && apt-get install -y python3-pip
+          pip3 install --upgrade --user pip
```

Comment:
Why do you `apt-get update` here before installing `pip`, but not for the other configurations?

And also:
Is it considered safe to mix the `pip` version distributed via the Ubuntu repository with a user-installed and updated version? I remember having troubles with that some time in the past (or a similar setup).

Reply:
I tried to remove the `apt-get update` and it did not work. Note that we use `precice/ci-ubuntu-1804` for this test and `precice/precice` for the other tests. I think that this is the reason. [`precice/ci-ubuntu-1804` runs `apt-get update` and then deletes the cache](https://github.com/precice/systemtests/blob/d39f4d89d69f18c9de9f8bd1f08633a68bcda9a2/precice/Dockerfile.Ubuntu1604.home#L6-L17) while [`precice/precice` does not delete the cache](https://github.com/precice/precice/blob/69edf625fe202e1006e4c91431ba923290e22e01/tools/releasing/packaging/docker/Dockerfile#L6-L7).

Reply:
Call `apt-get update` for all test now. 958aa3f

I also opened https://github.com/precice/precice/pull/927 resulting from this discussion. Thanks for the useful remark @ajaust!